src/Diagnostics/Samples/CSharp/CSharpAnalyzers/CSharpAnalyzers/StatefulAnalyzers/CodeBlockStartedAnalyzer.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace CSharpAnalyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class CodeBlockStartedAnalyzer : DiagnosticAnalyzer
+    {
+        #region Descriptor fields
+
+        internal static readonly LocalizableString Title = new LocalizableResourceString(nameof(Resources.CodeBlockStartedAnalyzerTitle), Resources.ResourceManager, typeof(Resources));
+        internal static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(Resources.CodeBlockStartedAnalyzerMessageFormat), Resources.ResourceManager, typeof(Resources));
+        internal static readonly LocalizableString Description = new LocalizableResourceString(nameof(Resources.CodeBlockStartedAnalyzerDescription), Resources.ResourceManager, typeof(Resources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticIds.CodeBlockStartedAnalyzerRuleId, Title, MessageFormat, DiagnosticCategories.Stateful, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+
+        #endregion
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCodeBlockStartAction<SyntaxKind>(UnusedParametersAnalyzer.CodeBlockStartAction);
```

Comment:
Having this be a lambda instead of a method in a separate class is less cognitive load. Given we are not capturing anything here, I think that would make it easier to understand the registration since you can see the nesting.